usaspending_api/idvs/v2/views/amounts.py

```diff
@@ -111,19 +111,30 @@ def fetch_idv_child_outlays(award_id: int, award_id_column) -> dict:
     if award_id_column != "award_id":
         award_id = re.sub(r"[']", r"''", award_id)
     sql = """
-    SELECT
-        COALESCE(sum(CASE WHEN sa.is_final_balances_for_fy = TRUE AND  THEN (COALESCE(faba.gross_outlay_amount_by_award_cpe,0)
-            + COALESCE(faba.ussgl487200_down_adj_pri_ppaid_undel_orders_oblig_refund_cpe, 0)
-            + COALESCE(faba.ussgl497200_down_adj_pri_paid_deliv_orders_oblig_refund_cpe, 0)) END), 0) AS total_outlay
-    FROM
-        financial_accounts_by_awards faba
-    INNER JOIN submission_attributes sa
-        ON faba.submission_id = sa.submission_id
-    INNER JOIN awards a
-        ON faba.award_id = a.id
-        AND a.date_signed >= '2019-10-01'
-    WHERE {award_id_sql}
-    """
+        with date_signed_outlay_amounts (award_id, last_period_total_outlay) as (
+            SELECT faba. award_id, COALESCE(sum(COALESCE(faba.gross_outlay_amount_by_award_cpe,0)
+                + COALESCE(faba.ussgl487200_down_adj_pri_ppaid_undel_orders_oblig_refund_cpe, 0)
+                + COALESCE(faba.ussgl497200_down_adj_pri_paid_deliv_orders_oblig_refund_cpe, 0)), 0) as last_period_total_outlay
+            FROM
+                financial_accounts_by_awards faba
+            INNER JOIN submission_attributes sa
+                ON faba.submission_id = sa.submission_id
+            INNER JOIN awards a
+                ON faba.award_id = a.id
+                AND a.date_signed >= '2019-10-01'
+            INNER JOIN transaction_normalized tn ON tn.id = a.earliest_transaction_id
+            WHERE sa.is_final_balances_for_fy AND sa.reporting_fiscal_year = tn.fiscal_year AND {award_id_sql}
```

Comment:
I see `award_id_sql` as a placeholder for format but then the call to format is `child_award_sql.format(award_id=award_id, award_id_column=award_id_column)` without `award_id_sql` at all. Is this filtering to the Award?

Reply:
`child_award_sql` is a different SQL statement that gets all the award_ids of the child awards, which are then plugged into this sql, same with `grandchild_award_sql`.

Reply:
And that is what I get for only looking at a single commit instead of the entire files changed... 🤦